code/game/objects/items/storage/belt.dm

```diff
@@ -929,4 +931,5 @@
 	. = ..()
 	var/obj/item/weapon/gun/new_gun = new /obj/item/weapon/gun/shotgun/double/marine(src)
 	new /obj/item/ammo_magazine/shotgun(src)
-	new_gun.on_enter_storage(src)
+	current_gun = new_gun
+	update_gun_icon()
```

Comment:
This should just use the on_enter storage stuff

Reply:
I figured it was better to keep the belt's behavior in the belt itself

Reply:
No?
Still pending